src/js/tools/pen.js

```diff
@@ -121,5 +124,20 @@ define(function (require, exports, module) {
     };
     util.inherits(PenTool, Tool);
 
+    /**
+     * Handler for keydown events, installed when the tool is active.
+     *
+     * @param {CustomEvent} event
+     */
+    PenTool.prototype.onKeyDown = function (event) {
+        var flux = this.getFlux(),
+            toolStore = flux.store("tool"),
+            detail = event.detail;
+  
+        if (toolStore.getVectorMode() && detail.keyCode === OS.eventKeyCode.ESCAPE) {
```

Comment:
I don't think it would be simpler, we already have the key down handler built into the tool code path. adding additional handlers seems less simple.